src/buildercore/bootstrap.py

```diff
@@ -654,21 +673,22 @@ def delete_stack(stackname):
     try:
         context = context_handler.load_context(stackname)
         terraform.destroy(stackname, context)
-        connect_aws_with_stack(stackname, 'cfn').delete_stack(stackname)
+        core.describe_stack(stackname).delete()
 
         def is_deleting(stackname):
             try:
                 return core.describe_stack(stackname).stack_status in ['DELETE_IN_PROGRESS']
-            except BotoServerError as err:
+            except botocore.exceptions.ClientError as err:
                 if err.message.endswith('does not exist'):
                     return False
                 raise # not sure what happened, but we're not handling it here. die.
         utils.call_while(partial(is_deleting, stackname), timeout=3600, update_msg='Waiting for AWS to finish deleting stack ...')
-        # remove_minion_key(stackname) # don't do this. prevents regular users deleting stacks
+        # don't do this. requires master server access and would prevent regular users deleting stacks
+        # remove_minion_key(stackname)
         keypair.delete_keypair(stackname) # deletes the keypair wherever it can find it (locally, remotely)
         delete_stack_file(stackname) # deletes the local cloudformation template
         delete_dns(stackname)
-
         LOG.info("stack %r deleted", stackname)
-    except BotoServerError as err:
+
```

Comment:
it's interesting how the errors turned from server faults to client faults in a new version :)

Reply:
I hadn't noticed, but yeah, now that you point it out ... And there is so little documentation around botocore and it's exceptions.